Enhancement and upgrade

jareddreyerss opened the pull request:
- Removed redundant dev build extension, don't know what its intention was so removed it.

- Fixed incorrect "feature name spacing", `SiteConfigExtension::class` is a `DataExtension` not a model.

- Updated `OnBeforeWrite()` calls to handle robots correctly and allow for testable scenarios.

jareddreyerss committed:
Enhancement and upgrade

Removed redundant dev build extension, don't know what its intention was so removed it.

Fixed incorrect feature namespacing. SiteconfigExtension is a dataextension not a model.

Updated onbeforewrite calls to handle robots correctly and allow for testable scenarios.

TheSceneman commented on tests/Extensions/SiteConfigExtensionTest.php:
```diff
@@ -0,0 +1,100 @@
+    public function robotsTxtStatusDataProvider(): Generator
+    {
+        yield 'Enabled' => [
+            'RobotosTxtStatus' => 1,
```
looks like a typo here, extra o in robots. Doesnt break anything but good to correct

TheSceneman commented on tests/Extensions/SiteConfigExtensionTest.php:
```diff
@@ -0,0 +1,100 @@
+        ];
+
+        yield 'Disabled' => [
+            'RobotosTxtStatus' => 0,
```
and here

TheSceneman commented on tests/Extensions/SiteConfigExtensionTest.php:
```diff
@@ -0,0 +1,100 @@
+    /**
+     * Helper so we can access our test robots.txt file path.
+     */
+    private function getRobotsTxtPath(SiteConfig $config): string
```
Be better if this was even more super descriptive as a function name. like `getRobotsTxtFilePath()` to really send home we're looking for a file here

TheSceneman commented on tests/Extensions/SiteConfigExtensionTest.php:
```diff
@@ -0,0 +1,100 @@
+        $this->siteConfig->onBeforeWrite();
+
+        // Get our robots.txt file
+        $path = $this->siteConfig->getPath() . '/robots.txt';
```
Should we be using the helper function below here?

TheSceneman reviewed (changes requested):
Awesome work this was a fun PR. Just a couple typos and tweaks in the test file

jareddreyerss committed:
Misc tidy ups and review feedback fixes

We weren't properly using Configurable to set the owner value of `$defaultRobotsDotTxtPath`. Thus have fixed everything to use proper scaffolding of this member.

Removed the `onBeforeWrite()` call in the test because when we call `SiteConfig::write()` it will automatically call `onBeforeWrite()`.

TheSceneman reviewed (approved):
LGTM
